src/tests/testClpMap.cc

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "testClpMap.h"
+#include "unitTestMain.h"
+
+#include "SquidConfig.h"
+
+#include <ctime>
+
+
+CPPUNIT_TEST_SUITE_REGISTRATION( testClpMap );
+
+class SquidConfig Config;
+
+void
+testClpMap::addData(testMap &m, int numElems, int base, testMap::Ttl ttl)
+{
+    for (int j = base; j < base + numElems; ++j ) {
+        CPPUNIT_ASSERT_EQUAL(true, m.add(std::to_string(j), j, ttl));
+    }
+}
+
+
+void
+testClpMap::setUp()
+{
+    squid_curtime = time(nullptr);
+}
+
+void
+testClpMap::PutGetDelete()
+{
+    testMap m(1024);
+    addData(m, 10);
+    CPPUNIT_ASSERT_EQUAL(static_cast<const int *>(nullptr), m.get("notthere"));
+    CPPUNIT_ASSERT_EQUAL(1, *(m.get("1")));
```

Comment:
This will _crash_ the executable if "1" is not actually there. We want the test to fail instead. I recommend refactoring the test code a little, so that you can say something like this here and in similar cases:

```suggestion
    CPPUNIT_ASSERT_EQUAL(1, Search(m, "1").value_or(0));
```

However, this is actually a weak test. We should test that the map returned not just any pointer to `1`, but the pointer we put there. That means refactoring to end up with something like this:

```suggestion
    CPPUNIT_ASSERT_EQUAL(item1, m.get("1"));
```

Reply:
But we are not putting in there a pointer, we are putting in there a value. ClpMap returns a pointer to the internal storage holding the value it copied in

Reply:
This change request was resolved without addressing it. The change request is valid -- only the second suggestion was flawed (as detailed above). I am unresolving this change request.

Reply:
> This change request was resolved without addressing it. The change request is valid -- only the second suggestion was flawed (as detailed above). I am unresolving this change request.

Oh, thanks. I had interpreted your concern as only being the second one

Reply:
All pointer dereferences are now guarded with a nonnull check, we should be good.